instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -29,21 +29,22 @@
 from opentelemetry import context, propagators, trace
 from opentelemetry.instrumentation.asgi.version import __version__  # noqa
 from opentelemetry.instrumentation.utils import http_status_to_canonical_code
+from opentelemetry.trace.propagation.textmap import DictGetter
 from opentelemetry.trace.status import Status, StatusCanonicalCode
 
 
-def get_header_from_scope(scope: dict, header_name: str) -> typing.List[str]:
-    """Retrieve a HTTP header value from the ASGI scope.
+# pylint:disable=arguments-differ
+class CarrierGetter(DictGetter):
+    def get(self, scope: dict, header_name: str) -> typing.List[str]:
```

Comment:
nit: should use carrier and key argument names and if you want, use doc string to explain that they are supposed to be asgi scope and header name.

Reply:
@nprajilesh ^